src/overlays/actors/ovl_En_Niw/z_en_niw.c

```diff
@@ -517,119 +555,122 @@ void EnNiw_Swimming(EnNiw* this, GlobalContext* globalCtx) {
     }
 
     this->actor.speedXZ = 2.0f;
-    if (this->actor.bgCheckFlags & 0x20) {
-        // still touching water
+    if (this->actor.bgCheckFlags & 0x20) { // touching water
         this->actor.gravity = 0.0f;
         if (this->actor.depthInWater > 15.0f) {
             this->actor.world.pos.y += 2.0f;
         }
-        if (this->unkTimer250 == 0) {
-            this->unkTimer250 = 30;
+        if (this->swimRippleTimer == 0) {
+            this->swimRippleTimer = 30;
             Math_Vec3f_Copy(&ripplePos, &this->actor.world.pos);
             ripplePos.y += this->actor.depthInWater;
 
             EffectSsGRipple_Spawn(globalCtx, &ripplePos, 100, 500, 30);
         }
-        if (this->actor.bgCheckFlags & 8) {
-            this->actor.velocity.y = 10.0f;
+        if (this->actor.bgCheckFlags & 8) { // hit a wall
+            this->actor.velocity.y = 10.0f; // fly up in straight line
             this->actor.speedXZ = 1.0f;
         }
     } else {
         this->actor.gravity = -2.0f;
-        if (this->actor.bgCheckFlags & 8) {
-            // has hit a wall
-            this->actor.velocity.y = 10.0f; // to the moon
+        if (this->actor.bgCheckFlags & 8) { // hit a wall
+            this->actor.velocity.y = 10.0f; // fly up in straight line
             this->actor.speedXZ = 1.0f;
             this->actor.gravity = 0.0f;
         } else {
             this->actor.speedXZ = 4.0f;
         }
 
-        if (this->actor.bgCheckFlags & 1) {
+        if (this->actor.bgCheckFlags & 1) { // hit floor
             this->actor.gravity = -2.0f;
-            this->unkTimer254 = 100;
-            this->unkTimer250 = 0;
+            this->runAwayTimer = 100;
+            this->swimRippleTimer = 0;
             this->actor.velocity.y = 0.0f;
             if (!this->isStormActive) {
                 EnNiw_SetupRunAway(this);
             } else {
-                this->unknownState28E = 3;
+                this->niwState = NIW_STATE_ANGRY3;
                 this->actionFunc = EnNiw_CuccoStorm;
             }
         }
     }
-    func_80891320(this, globalCtx, 2);
+
+    EnNiw_AnimateWingHead(this, globalCtx, NIW_ANIMATION_PECKING_AND_WAVING);
 }
 
+/**
+ *  Possible Fake Match: the weird way this state is set
+ */
 void EnNiw_Trigger(EnNiw* this, GlobalContext* globalCtx) {
-    s32 value;
+    s32 state;
     if (1) {
-        value = 1;
+        state = NIW_STATE_ANGRY1;
```

Comment:
This kind of comment looks like a documentation comment, which seems a bit weird just for this

```c
void EnNiw_Trigger(EnNiw* this, GlobalContext* globalCtx) {
    s32 state;

    // Possible Fake Match: the weird way this state is set
    if (1) {
        state = NIW_STATE_ANGRY1;
```